### PR TITLE
Improve AspNetCore.Demo with example of how to load assemblies

### PR DESCRIPTION
**Pull request by haacked**
This PR provides a demo for the issue reported in https://github.com/ashmind/mirrorsharp/issues/101.

__Repro Steps:__

1. Add a new `netstandard2.1` class library with a couple of interfaces.
2. Set the MirrorSharp to script mode and pass in an interface as the host type.
3. Use the code in https://github.com/ashmind/mirrorsharp/issues/101 to add all referenced assemblies.
4. Run the demo project and add code that uses the host object.

Expected:
No compilation errors.

Actual:
A compilation error that suggests adding a reference to `System.Runtime` which doesn't make sense for .NET Core on Mac OS.

Screen Shot 2020-08-25 at 1 48 43 PM

### CODE REVIEW

**Comment by haacked** [comment]
I've updated this PR with an example that shows how to properly load assemblies in the case you want to use your own custom type as a host object. This required I add a small little class library to hold the host object.

@ashmind I think this change will help others who run into this problem.

**Comment by ashmind** [comment]
Thanks, that's a good idea! I'll add some cosmetic comments but overall it makes sense.

File: AspNetCore.Demo.Library/Interfaces.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace AspNetCore.Demo.Library
+{
+    public interface IScriptGlobals
```
Comment (ashmind) [inline AspNetCore.Demo.Library/Interfaces.cs:6]: I generally try to have file per type, so I would split those into separate files.

File: AspNetCore.Demo/Startup.cs [hunk 2]
```diff
@@ -25,10 +31,28 @@ public class Startup {
                 "/mirrorsharp",
                 new MirrorSharpOptions {
                     SelfDebugEnabled = true,
-                    IncludeExceptionDetails = true
+                    IncludeExceptionDetails = true,
```
Comment (ashmind) [inline AspNetCore.Demo/Startup.cs:34]: I normally avoid dangling commas, not sure if there is a an editorconfig setting for it...

File: AspNetCore.Demo/Startup.cs [hunk 3]
```diff
@@ -25,10 +31,28 @@ public class Startup {
...
                 }
+                .SetupCSharp(o => {
+                    o.AddMetadataReferencesFromFiles(GetAllReferencePaths().ToArray());
+                    o.SetScriptMode(true, typeof(IScriptGlobals));
+                })
                 .EnableFSharp()
             ));
         }
+
+        static IEnumerable<string> GetAllReferencePaths() {
```
Comment (ashmind) [inline AspNetCore.Demo/Startup.cs:44]: I generally use explicit access levels.

**Comment by ashmind** [comment]
I don't mind doing changes myself btw, will just take slightly more time.

File: AspNetCore.Demo.Library/Interfaces.cs (hunk 1 above)
Comment (haacked) [reply AspNetCore.Demo.Library/Interfaces.cs]: Fixed!

**Comment by haacked** [comment]
Thanks for the review! I think I addressed all of your comments.

**Comment by ashmind** [comment]
Great work, thanks!

**Pull request merged**